llvm/lib/Object/DXContainer.cpp

```diff
@@ -279,14 +276,21 @@ Error DirectX::RootSignature::parse(StringRef Data) {
       support::endian::read<uint32_t, llvm::endianness::little>(Current);
   Current += sizeof(uint32_t);
 
-  uint32_t FValue =
-      support::endian::read<uint32_t, llvm::endianness::little>(Current);
+  Flags = support::endian::read<uint32_t, llvm::endianness::little>(Current);
   Current += sizeof(uint32_t);
 
-  if (!dxbc::RootSignatureValidations::isValidRootFlag(FValue))
-    return validationFailed("unsupported root signature flag value read: " +
-                            llvm::Twine(FValue));
-  Flags = FValue;
+  assert(Current == Begin + RootParametersOffset);
+
+  ParametersHeaders.Data = Data.substr(
+      RootParametersOffset, NumParameters * sizeof(dxbc::RootParameterHeader));
+
+  ParameterSpaceOffset =
+      RootParametersOffset + NumParameters * sizeof(dxbc::RootParameterHeader);
+  size_t ParameterSpaceEnd =
+      Data.size() - ((NumStaticSamplers == 0) ? 0 : StaticSamplersOffset);
```

Comment:
Is this correct?  Seems strange to subtract `StaticSamplersOffset`.

Firstly, since we don't support static samplers, I'd be tempted to just have `assert(NumStaticSamplers == 0)` and then worry about integrating static samplers into this calculation when we get to it.

I think that the idea is that ParameterSpaceEnd is either everything up to the first static sampler, or to the end of the data if there are no static samplers.  So, I think that this would be correct:

```c++
  size_t ParameterSpaceEnd = (NumStaticSamplers == 0) ? Data.size() : StaticSamplersOffset;
```